libethereum/Client.cpp

```diff
@@ -782,11 +786,15 @@ LocalisedLogEntries Client::logs(LogFilter const& _f) const
 		int total = 0;
 #endif
 		// check block bloom
-		if (_f.matches(m_bc.info(h).logBloom))
-			for (TransactionReceipt receipt: m_bc.receipts(h).receipts)
+		auto info = m_bc.info(h);
+		auto receipts = m_bc.receipts(h).receipts;
+		if (_f.matches(info.logBloom))
+			for (size_t i = 0; i < receipts.size(); i++)
 			{
+				TransactionReceipt receipt = receipts[i];
 				if (_f.matches(receipt.bloom()))
 				{
+					auto sha3 = transaction(info.hash, i).sha3();
```

Comment:
better to use e.g. `h` rather than `sha3`, since the latter is a well-used symbol for the SHA3 function in `dev::`.